### PR TITLE
Fix --project help

### LINKED ISSUE #2867: tsconfig.json: is multiple build configurations possible?
What's the best way to accomplish multiple build profiles with tsconfig? I want to have builds that specifically target different platforms like node/browser etc..

``` javascript
{
   es5: {
      compilerOptions: {
        target: 'es5',
      },
      files: [
        'polyfill.ts',
        'index.ts'
      ]
   }
   es6: {
      compilerOptions: {
        target: 'es6',
      },
      files: [
        'index.ts'
      ]
   }
   nodees5: {
      compilerOptions: {
        target: 'es5',
      },
      files: [
        'polyfill.ts',
        'index.ts',
        'nodexyz.ts'
      ]
   }
   nodees6: {
      compilerOptions: {
        target: 'es6',
      },
      files: [
        'index.ts',
        'nodexyz.ts'
      ]
   }
}
```

### PR DESCRIPTION
**Pull request by unional**
Fixes #2867

https://github.com/Microsoft/TypeScript/issues/2867#issuecomment-205451109

One concern is that the description is getting quite long.

Any suggestion?

### CODE REVIEW

File: src/compiler/diagnosticMessages.json [hunk 1]
```diff
@@ -2573,6 +2573,10 @@
         "category": "Message",
         "code": 6039
     },
+    "DIRECTORY_OR_FILE": {
```
Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: remove the `_` from the message. so it should be `FILE or DIRECTORY`

File: src/compiler/diagnosticMessages.json [hunk 2]
```diff
@@ -2513,7 +2513,7 @@
         "category": "Message",
         "code": 6019
     },
-    "Compile the project in the given directory.": {
+    "Compile the project in the given directory, using 'tsconfig.json' or the specified config file.": {
```
Comment (DanielRosenwasser) [inline src/compiler/diagnosticMessages.json]: Compile a project given the path to its configuration file, or to a folder with a 'tsconfig.json'.

Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: how about:
`Compile the project defined by the config file. Argument can be a full path to a config file, or a directory path to a directory containing 'tsconfig.json'`

Comment (mhegazy) [inline src/compiler/diagnosticMessages.json]: scrap that, i like @DanielRosenwasser's comment better.

**Comment by unional** [comment]
Oh, I just realized I did `FILE OR DIRECTORY` instead of `FILE or DIRECTORY`.
Is that ok?

**Comment by mhegazy** [comment]
would work as well.

**Pull request merged**